### ISSUE TITLE
typeof(T).Size is not constant anymore

### ISSUE THREAD

**Issue by disarray2077** [beefytech/Beef#1076]
I guess after the issue https://github.com/beefytech/Beef/issues/1046 was fixed, the `typeof(T).Size` stopped being recognized as constant, because now when it's used as an array size the IDE complains about it not being a `constant length`

**Comment by bfiete** [comment]
Can you post a code example that fails?

This seems to work, as expected:
```C#
void Test<T>()
{
	const int size = typeof(T).Size;
	int[typeof(T).Size] arr;
}
```

**Comment by disarray2077** [comment]
It is being used like this:
```
T* data = new uint8[typeof(T).Size]* (0,);
```
_(This code is simplified but I guess this should reproduce the issue)_

**Comment by bfiete** [comment]
Fixed at 90a51834bc6fe0a56a6f1da398684a7346284a78
